brave/src/main/java/brave/propagation/B3SingleFormat.java

```diff
@@ -50,7 +74,7 @@ static String writeB3SingleFormat(TraceContext context) {
     return new String(result, 0, pos);
   }
 
-  static @Nullable TraceContextOrSamplingFlags maybeB3SingleFormat(String b3) {
+  @Nullable public static TraceContextOrSamplingFlags parseB3SingleFormat(String b3) {
```

Comment:
this part, changing to CharSequence is one idea.

PS @llinder @michaelsembwever I think we should switch cassandra to use this single-header propagation format, since it is a good time anyway (as currently trace propagation is turned off in brave-cassandra!)